packages/fxa-auth-server/test/local/payments/customer-locations.js

```diff
@@ -0,0 +1,236 @@
+const sinon = require('sinon');
+
+const STATES_LONG_NAME_TO_SHORT_NAME_MAP = require('../../../lib/payments/states-long-name-to-short-name-map.json');
+const { deepCopy } = require('./util');
+const { mockLog } = require('../../mocks');
+const {
+  CustomerLocations,
+} = require('../../../lib/payments/customer-locations');
+
+const sandbox = sinon.createSandbox();
+
+const mockPayPalCustomer = {
+  id: 'cus_123',
+  address: {
+    country: 'CA',
+  },
+};
+const mockUid = 'abc123';
+
+describe('CustomerLocations', () => {
+  let mockStripeHelper;
+  let customerLocations;
+  const mockLocationResults = [
+    {
+      uid: mockUid,
+      state: 'Saxony-Anhalt',
+      country: 'Germany',
+      count: 2,
+    },
+    {
+      uid: mockUid,
+      state: 'British Columbia',
+      country: 'Canada',
+      count: 6,
+    },
+  ];
+  mockLog.debug = sandbox.fake.returns({});
+
+  beforeEach(() => {
+    mockStripeHelper = {
+      fetchCustomer: () => mockPayPalCustomer,
+    };
+    customerLocations = new CustomerLocations(mockLog, mockStripeHelper);
+  });
+
+  afterEach(() => {
+    sandbox.reset();
+  });
+
+  describe('constructor', () => {
+    it('sets log and Stripe helper', () => {
+      assert.strictEqual(customerLocations.log, mockLog);
+      assert.strictEqual(customerLocations.stripeHelper, mockStripeHelper);
+    });
+  });
+
+  describe('getBestLocationForPayPalUser', () => {
+    it('returns unknown country and state when no country is found in Stripe', async () => {
+      mockStripeHelper.fetchCustomer = sandbox.fake.resolves({
+        ...deepCopy(mockPayPalCustomer),
+        address: null,
+      });
+      const expected = {
+        uid: mockUid,
+        state: 'unknown',
+        country: 'unknown',
+        count: 1,
+      };
+      const result = await customerLocations.getBestLocationForPayPalUser(
+        mockUid,
+        mockLocationResults
+      );
+      assert.deepEqual(result, expected);
+    });
+    it('returns Stripe country and unknown state when no inferred locations are provided', async () => {
+      const expected = {
+        uid: mockUid,
+        state: 'unknown',
+        country: mockPayPalCustomer.address.country,
+        count: 1,
+      };
+      const result = await customerLocations.getBestLocationForPayPalUser(
+        mockUid,
+        []
+      );
+      assert.deepEqual(result, expected);
+    });
+    it('returns null when the location for the Stripe country is not needed', async () => {
+      mockStripeHelper.fetchCustomer = sandbox.fake.resolves({
+        ...deepCopy(mockPayPalCustomer),
+        address: {
+          country: 'DE',
+        },
+      });
+      const result = await customerLocations.getBestLocationForPayPalUser(
+        mockUid,
+        mockLocationResults
+      );
+      assert.deepEqual(result, null);
+    });
+    it('returns Stripe country and unknown state when no matching country is found', async () => {
+      const otherCountry = 'US';
```

Comment:
Curious as to why this is named `otherCountry`

Reply:
It's another, different country relative to the countries in `mockLocationResults`. I could call it something less confusing like `nonmatchingCountry` though.